File: modules/waves/waves.c
#include<unistd.h>
#include<time.h>
#include<sys/time.h>
#include<stdio.h>
#include<stdlib.h>
#include<stdint.h>
#include<sys/resource.h>
#include<wiringPi.h>

#define GPIO_TRIG 4
#define GPIO_ECHO 5

int main(void){
	unsigned int start,stop;
	int loop=0,count;
	float f;

	printf("starting hc-sr04 test \n");
	int status=wiringPiSetup();
	if(status!=0){
		printf("fail to wiringpi setup\n");
		return 0;
	}
	pinMode(GPIO_TRIG,OUTPUT);
	pinMode(GPIO_ECHO,INPUT);

	digitalWrite(GPIO_TRIG,LOW);
	delay(500);
	printf("st loop\n");
	while(1){
		printf("measure distance... once per 2 seconds\n");
		delay(2000);
		digitalWrite(GPIO_TRIG,HIGH);
		delayMicroseconds(10);
		digitalWrite(GPIO_TRIG,LOW);
		while(digitalRead(GPIO_ECHO) == LOW){
			//printf("loop1\n");
		}
		start=micros();
		while(digitalRead(GPIO_ECHO) == HIGH){
			//printf("loop2\n");
		}
		stop=micros();
		double dis = 34209*(stop-start)/2000000.0;
		printf("dis : %fcm\n",dis);
		//if(loop++ == 100)break;
	}
	return 0;
}

File: modules/waves/waves.py
import RPi.GPIO as GPIO
import time

GPIO_TRIG=23
GPIO_ECHO=24
GPIO.setmode(GPIO.BCM)
print("distance meassurement in progress")

GPIO.setup(GPIO_TRIG,GPIO.OUT)
GPIO.setup(GPIO_ECHO,GPIO.IN)
GPIO.output(GPIO_TRIG,False)

print("waiting for sensor to settle")
time.sleep(2)
GPIO.output(GPIO_TRIG,True)
time.sleep(0.00001)

GPIO.output(GPIO_TRIG,False)
while True:
	while GPIO.input(GPIO_ECHO)==0:
		pulse_start=time.time()
	while GPIO.input(GPIO_ECHO)==1:
		pulse_end=time.time()
	pulse_duration = pulse_end - pulse_start
	distance=pulse_duration*17150
	distance = round(distance,2)
	print("distance : ",distance,"cm")
	time.sleep(1)
	GPIO.output(GPIO_TRIG,False)

GPIO.cleanup()
